python/lsst/ip/isr/isrTask.py

```diff
@@ -1860,9 +1894,38 @@ def overscanCorrection(self, ccdExposure, amp):
         overscanResults = self.overscan.run(ccdExposure, amp)
 
         metadata = ccdExposure.getMetadata()
-        ampNum = amp.getName()
-        metadata[f"ISR_OSCAN_LEVEL{ampNum}"] = overscanResults.overscanMean
-        metadata[f"ISR_OSCAN_SIGMA{ampNum}"] = overscanResults.overscanSigma
+        ampName = amp.getName()
+
+        keyBase = "LSST ISR OVERSCAN"
+        # Updated quantities
+        if isinstance(overscanResults.overscanMean, float):
+            # Serial overscan correction only:
+            metadata[f"{keyBase} SERIAL MEAN {ampName}"] = overscanResults.overscanMean
+            metadata[f"{keyBase} SERIAL MEDIAN {ampName}"] = overscanResults.overscanMedian
+            metadata[f"{keyBase} SERIAL STDEV {ampName}"] = overscanResults.overscanSigma
+
+            metadata[f"{keyBase} RESIDUAL SERIAL MEAN {ampName}"] = overscanResults.residualMean
+            metadata[f"{keyBase} RESIDUAL SERIAL MEDIAN {ampName}"] = overscanResults.residualMedian
+            metadata[f"{keyBase} RESIDUAL SERIAL STDEV {ampName}"] = overscanResults.residualSigma
+        elif isinstance(overscanResults.overscanMean, tuple):
+            # Both serial and parallel overscan have run:
+            metadata[f"{keyBase} SERIAL MEAN {ampName}"] = overscanResults.overscanMean[0]
+            metadata[f"{keyBase} SERIAL MEDIAN {ampName}"] = overscanResults.overscanMedian[0]
+            metadata[f"{keyBase} SERIAL STDEV {ampName}"] = overscanResults.overscanSigma[0]
+
+            metadata[f"{keyBase} PAR MEAN {ampName}"] = overscanResults.overscanMean[1]
+            metadata[f"{keyBase} PAR MEDIAN {ampName}"] = overscanResults.overscanMedian[1]
+            metadata[f"{keyBase} PAR STDEV {ampName}"] = overscanResults.overscanSigma[1]
+
+            metadata[f"{keyBase} RESIDUAL SERIAL MEAN {ampName}"] = overscanResults.residualMean[0]
+            metadata[f"{keyBase} RESIDUAL SERIAL MEDIAN {ampName}"] = overscanResults.residualMedian[0]
+            metadata[f"{keyBase} RESIDUAL SERIAL STDEV {ampName}"] = overscanResults.residualSigma[0]
+
+            metadata[f"{keyBase} RESIDUAL PAR MEAN {ampName}"] = overscanResults.residualMean[1]
+            metadata[f"{keyBase} RESIDUAL PAR MEDIAN {ampName}"] = overscanResults.residualMedian[1]
+            metadata[f"{keyBase} RESIDUAL PAR STDEV {ampName}"] = overscanResults.residualSigma[1]
+        else:
+            self.log.warning("Unknown length of overscan values; none added to header.")
```

Comment:
The warning message seems misleading. This warning will be emitted if `overscanResults.overscanMean` is neither a `float` nor a `tuple` but this message says it's of unknown length. It should be something like "Overscan values are of unacceptable type; none added to header.")